programs/clearing_house/src/lib.rs

```diff
@@ -747,6 +749,210 @@ pub mod clearing_house {
         Ok(())
     }
 
+    #[allow(unused_must_use)]
+    #[access_control(
+        exchange_not_paused(&ctx.accounts.state)
+    )]
+    pub fn remove_liquidity<'info>(
+        ctx: Context<OpenPosition>,
+        market_index: u64,
+    ) -> Result<()> {
+        let user = &mut load_mut(&ctx.accounts.user)?;
+        let remaining_accounts_iter = &mut ctx.remaining_accounts.iter().peekable();
+
+        let market_map = MarketMap::load(
+            &get_writable_markets(market_index),
+            &get_market_oracles(market_index, &ctx.accounts.oracle),
+            remaining_accounts_iter,
+        )?;
+        let mut market = market_map.get_ref_mut(&market_index)?;
+
+        let position_index = get_position_index(&user.positions, market_index)?;
+        let lp_position = &mut user.positions[position_index];
+        
+        // todo: impl not burning all the tokens 
+        // todo: settle fcn with no burning 
+        let lp_tokens_to_burn = lp_position.lp_tokens; 
+        let total_lp_tokens = market.amm.sqrt_k;
+        
+        validate!(
+            lp_position.lp_tokens <= lp_tokens_to_burn,
+            ErrorCode::InsufficientLPTokens,
+            "Trying to burn more lp tokens than the user has",
+        )?;
+
+        if lp_tokens_to_burn == 0 {
+            return Ok(());
+        }
+
+        let lp_share = |v: i128| -> Result<i128> {
```

Comment:
i would make a new `lp` in the math folder which has this function. you'll probably need to reuse it in the margin functions